tools/envvars-ExecStartPre.sh.in

```diff
@@ -89,15 +100,47 @@ if [ -x /usr/bin/systemd-detect-virt ]; then
     VIRT_HYPERVIZOR="$(/usr/bin/systemd-detect-virt -v)" || VIRT_HYPERVIZOR=""
 fi
 
-cat << EOF > "$F"
+ENV_HOSTNAME="$(hostname)" || ENV_HOSTNAME=""
+ENV_DOMAINNAME="$(domainname | grep -v '(none)')" || ENV_DOMAINNAME=""
+ENV_DNSDOMAINNAME="$(dnsdomainname | grep -v '(none)')" || ENV_DNSDOMAINNAME=""
+# A comma-separated list (required for SQL magic, usable in shell) of
+# current non-loopback IP addresses, IPv4 and IPv6 alike, without masks
+ENV_IPADDRS="$(ip addr show | egrep '^[[:blank:]]*inet6?[[:blank:]]' | awk '{print $2}' | egrep -v '^(127\..*|::1/128)$' | sed 's,/[0-9]*$,,' | tr '\n' ',' | sed 's/,$//')" || ENV_IPADDRS=""
+
+RES=255
+case "$OUTFMT" in
+    env) cat << EOF > "$OUTFILE"
 OSIMAGE_BASENAME='$OSIMAGE_BASENAME'
-SPOOL_DIR='${SPOOL}'
 HARDWARE_CATALOG_NUMBER='${HARDWARE_CATALOG_NUMBER}'
 HARDWARE_SPEC_REVISION='${HARDWARE_SPEC_REVISION}'
 HARDWARE_SERIAL_NUMBER='${HARDWARE_SERIAL_NUMBER}'
 HARDWARE_UUID='${HARDWARE_UUID}'
 VIRT_HYPERVIZOR='${VIRT_HYPERVIZOR}'
```

Comment:
the previous fallback ' || VIRT_CONTAINER=""' empties both VIRT_* vars, as per manpage:
EXIT STATUS
       If a virtualization technology is detected, 0 is returned, a non-zero code otherwise.

So we end up with "" instead of "none".
These 2 values are crucial for some processing, plz fix!

Reply:
Orthogonal to this PR, but makes sense, tested true - will do :)